src/v1/sources/shopify/webhookTransformations/serverSideUtlis.js

```diff
@@ -73,22 +90,28 @@ const getCartToken = (event) => event?.cart_token || null;
  */
 const setAnonymousId = async (message, event, metricMetadata) => {
   const anonymousId = getAnonymousIdFromAttributes(event);
+  const cartToken = getCartToken(event);
+  const cartTokenHash = cartToken ? uuidv5(cartToken) : null;
   if (isDefinedAndNotNull(anonymousId)) {
     message.anonymousId = anonymousId;
-  } else {
-    // if anonymousId is not present in note_attributes or note_attributes is not present, query redis for anonymousId
-    const cartToken = getCartToken(event);
-    if (cartToken) {
-      const redisData = await RedisDB.getVal(`pixel:${cartToken}`);
-      if (redisData?.anonymousId) {
-        message.anonymousId = redisData.anonymousId;
-      }
+  }
+  // if anonymousId is not present in note_attributes or note_attributes is not present, query redis for anonymousId
+  // when cart_token is present
+  else if (cartToken) {
+    const redisData = await RedisDB.getVal(`pixel:${cartToken}`);
+    if (redisData?.anonymousId) {
+      message.anonymousId = redisData.anonymousId;
     } else {
-      stats.increment('shopify_pixel_cart_token_not_found_server_side', {
-        source: metricMetadata.source,
-        writeKey: metricMetadata.writeKey,
-      });
+      // if anonymousId is not present in note_attributes or redis, generate a new anonymousId
+      // the anonymousId will be generated by hashing the cart_token using uuidv5
+      // this hash will be present in the traits object as cart_token_hash
+      message.anonymousId = cartTokenHash;
```

Comment:
Do we have metric to track events that are missed from identity stitching ?

Reply:
Added one, similar to one we have already in legacy app
`shopify_redis_no_val`

Reply:
rename stat to `shopify_id_stitch_gaps`

Reply:
done,